src/main/java/ch/njol/skript/events/EvtFurnace.java

```diff
@@ -0,0 +1,101 @@
+package ch.njol.skript.events;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.aliases.ItemType;
+import ch.njol.skript.lang.Literal;
+import ch.njol.skript.lang.SkriptEvent;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.registrations.Classes;
+import org.bukkit.event.Event;
+import org.bukkit.event.inventory.FurnaceBurnEvent;
+import org.bukkit.event.inventory.FurnaceExtractEvent;
+import org.bukkit.event.inventory.FurnaceSmeltEvent;
+import org.bukkit.event.inventory.FurnaceStartSmeltEvent;
+import org.jetbrains.annotations.Nullable;
+
+public class EvtFurnace extends SkriptEvent {
+
+	static {
+		Skript.registerEvent("Smelt", EvtFurnace.class, FurnaceSmeltEvent.class,
+				"[furnace] [ore] smelt[ed|ing] [of %-itemtypes%]",
+				"[furnace] smelt[ed|ing] of ore")
+			.description("Called when a furnace smelts an item in its <a href='expressions.html#ExprFurnaceSlot'>ore slot</a>.")
+			.examples(
+				"on smelt:",
+				"on smelt of raw iron:",
+					"\tbroadcast smelted item",
+					"\tset smelt item to iron block"
+			)
+			.since("1.0, INSERT VERSION (specific item)");
+		Skript.registerEvent("Fuel Burn", EvtFurnace.class, FurnaceBurnEvent.class, "[furnace] fuel burn[ing] [of %-itemtypes%]")
+			.description("Called when a furnace burns an item from its <a href='expressions.html#ExprFurnaceSlot'>fuel slot</a>.")
+			.examples(
+				"on fuel burning:",
+					"\tbroadcast fuel burned",
+					"\tif fuel burned is coal:",
+						"\t\tadd 20 seconds to burn time"
+			)
+			.since("1.0, INSERT VERSION (specific item)");
+		Skript.registerEvent("Furnace Extract", EvtFurnace.class, FurnaceExtractEvent.class, "furnace extract[ion] [of %-itemtypes%]")
```

Comment:
```suggestion
		Skript.registerEvent("Furnace Item Extract", EvtFurnace.class, FurnaceExtractEvent.class, "furnace [item] extract[ion] [of %-itemtypes%]")
```
maybe there's some other word for extract that you could add as synonym? extract isn't instantly clear for me